sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryDiagnosticsHandler.java

```diff
@@ -51,11 +53,17 @@ public void handleDiagnostics(CosmosDiagnosticsContext diagnosticsContext, Conte
 
         this.ensureInitialized();
 
+        ClientTelemetry clientTelemetrySnapshot = this.telemetryRef.get();
+        if (clientTelemetrySnapshot == null) {
```

Comment:
Same here? Is this condition possible? Is it intentional to just return from here? how will it affect the workflow of customers?

Reply:
Yes, it is intentional. Only way to get here I can see are race conditions - remaining requests while client gets closed. Worst case we lose client-telemetry and/or metrics - nothing critical.